lib/ansible/modules/storage/purestorage/purefa_hg.py

```diff
@@ -99,18 +124,48 @@ def make_hostgroup(module, array):
     changed = True
 
     if not module.check_mode:
-        host = array.create_hgroup(module.params['hostgroup'])
+        array.create_hgroup(module.params['hostgroup'])
         if module.params['host']:
             array.set_hgroup(module.params['hostgroup'], hostlist=module.params['host'])
         if module.params['volume']:
-            for v in module.params['volume']:
-                array.connect_hgroup(module.params['hostgroup'], v)
+            for vol in module.params['volume']:
+                array.connect_hgroup(module.params['hostgroup'], vol)
     module.exit_json(changed=changed)
 
 
 def update_hostgroup(module, array):
     changed = False
-    hostgroup = module.params['hostgroup']
+    hgroup = get_hostgroup(module, array)
+    volumes = array.list_hgroup_connections(module.params['hostgroup'])
+    if module.params['state'] == "present":
+        if module.params['host']:
+            new_hosts = list(set(module.params['host']).difference(hgroup['hosts']))
+            if new_hosts:
+                array.set_hgroup(module.params['hostgroup'], addhostlist=new_hosts)
```

Comment:
I don't know the `purestorage.FlashArray` API behind the `array` type, but was curious if there a potential error condition or unhandled exception that could occur here in the event the attempt to `set_hgroup` fails? (same question for various `set_hgroup`, `connect_hgroup`, and `set_hgroup` calls in this patch).

Reply:
Good point. I'll add try/except where necessary